front-end/src/util/sorting.js

```diff
@@ -1,10 +1,22 @@
 export const default_sort = (data) => {
     return data
-        .sort((a, b) => {
-            return b['mentees'].length - a['mentees'].length})
+        .sort((craftsperson1, craftsperson2) => {return craftsperson2['mentees'].length - craftsperson1['mentees'].length})
         .map((craftsperson) => {
-            if(craftsperson['mentor'] != null)
-                return 'Craftsperson: ' + craftsperson['firstName'] + ' ' + craftsperson['lastName'] + ' - Mentor: ' + craftsperson['mentor'].split(' ')[0] + ' ' + craftsperson['mentor'].split(' ')[1];
-            return 'Craftsperson: ' + craftsperson['firstName'] + ' ' + craftsperson['lastName'];
-        });
-};
\ No newline at end of file
+           const person = 'Craftsperson: ' + craftsperson['firstName'] + ' ' + craftsperson['lastName'];
+
+           let mentor = '-';
+           if(craftsperson['mentor'] != null) {
+                   mentor = craftsperson.mentor.firstName + ' ' + craftsperson.mentor.lastName;
+            }
+
+           let mentees = '-';
+           if(craftsperson.mentees.length > 0) {
+               mentees = craftsperson.mentees
+                   .map(m => m.firstName + ' ' + m.lastName)
+                   .reduce((m1, m2) => m1 + ', ' + m2);
```

Comment:
The variable names make unclear the purpose of the `reduce` call

Reply:
Maybe you could rewrite it to use a `join()`? Not everyone is skilled with functional programming and I guess it would be easier to understand at a glance.